article/app/controllers/ArticleController.scala

```diff
@@ -234,15 +260,29 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 }
 
 object ParseBlockId extends RegexParsers {
-  def apply(input: String): Option[String] = {
-    def withParser: Parser[Unit] = "with:" ^^ { _ => () }
-    def block: Parser[Unit] = "block-" ^^ { _ => () }
-    def id: Parser[String] = "[a-zA-Z0-9]+".r
-    def expr: Parser[String] = withParser ~> block ~> id
+
```

Comment:
By the way, parsing with regex like this is pretty slow. Hard to say without profiling whether it's worth worrying about, but rewriting this parser using [parboiled](https://github.com/sirthias/parboiled2) might be a quick performance win.